client/dimensions_api.go

```diff
@@ -0,0 +1,151 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	logKeys "github.com/ONSdigital/dp-dimension-importer/common"
+	"github.com/ONSdigital/dp-dimension-importer/model"
+	"github.com/ONSdigital/go-ns/log"
+)
+
+const (
+	unmarshallingErr      = "Unexpected error while unmarshalling response"
+	unexpectedAPIErr      = "Unexpected error returned when calling Import API"
+	hostConfigMissingErr  = "DimensionsClient requires an API host to be configured"
+	marshalDimensionErr   = "Unexpected error while marshalling dimenison"
+	instanceIDRequiredErr = "instanceID is required but is empty"
+
+	getDimensionsURIFMT  = "%s/instances/%s/dimensions"
+	getDimensionsSuccess = "Import-API Get Dimensions success"
+	getDimensionsErr     = "Get dimensions returned error status"
+
+	createPutNodeIDReqErr  = "Unexpecter error creating request struct"
+	putDimensionNodeIDURI  = "%s/instances/%s/dimensions/%s/node_id/%s"
+	putDimNodeIDSuccessLog = "Import-API PUT dimension node_id success"
+	putDimNodeIDReqErr     = "Error sending set Dimension node_id request"
+	putDimNodeIDErr        = "Set Dimension node_id returned error status"
+	dimensionNilErr        = "Dimension is required but was nil"
+)
+
+// HTTPClient interface for making HTTP requests.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var httpCli HTTPClient = &http.Client{}
+
+// Host the host to get the dimensions from.
+var Host string
+
+// httpGet abstraction around http.Get to simplify testing / mocking.
+var httpGet = http.Get
+
+// respBodyReader abstraction around ioutil.ReadAll to simplify testing / mocking.
+var respBodyReader = ioutil.ReadAll
+
+// ImportAPI provides methods for getting dimensions for a given instanceID and updating the node_id of a specific dimension.
+type ImportAPI struct{}
```

Comment:
Done - Iv moved all the package level vars used in the struct functions to be fields of the struct.